solutions/ha-nva/layer-7-ingress/nested/storekeyinkeyvault.json

```diff
@@ -0,0 +1,183 @@
+{
+    // This template is adapted from: https://azure.microsoft.com/resources/templates/201-deployment-script-ssh-key-gen/
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]",
+            "metadata": {
+                "description": "The location for the script resource."
+            }
+        },
+        "newGuid": {
+            "type": "string",
+            "defaultValue": "[newGuid()]",
+            "metadata": {
+                "description": "forceUpdateTag property, used to force the execution of the script resource when no other properties have changed."
+            }
+        },
+        "passPhrase": {
+            "type": "securestring",
+            "minLength": 6,
+            "maxLength": 64,
+            "metadata": {
+                "description": "Passphrase used when generating the key pair."
+            }
+        },
+        "vaultName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the keyVault to store the keys in."
+            }
+        },
+        "secretName": {
+            "type": "string",
+            "defaultValue": "privateKey",
+            "metadata": {
+                "description": "The name of the secret in keyVault to store the keys in."
+            }
+        },
+        "vaultResourceGroup": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().name]",
+            "metadata": {
+                "description": "The resourceGroup for the keyVault."
+            }
+        },
+        "vaultSubscriptionId": {
+            "type": "string",
+            "defaultValue": "[subscription().subscriptionId]",
+            "metadata": {
+                "description": "The subscriptionId for the keyVault."
+            }
+        },
+        "_artifactsLocation": {
+            "type": "string",
+            "metadata": {
+                "description": "The base URI where artifacts required by this template are located. When the template is deployed using the accompanying scripts, a private location in the subscription will be used and this value will be automatically generated."
+            },
+            "defaultValue": "[deployment().properties.templateLink.uri]"
```

Comment:
I believe ARM TTK is complaining about this, but have not dug in.

```
    [-] artifacts parameter (14 ms)                                                                                                                                                       The _artifactsLocation and _artifactsLocationsSasToke parameters in "test.json" must not have a defaulValue in a nested template.   
```

Reply:
Removed the default values.  The template I copied from was top level, but this makes sense.